components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -573,6 +574,7 @@ void AIChatTabHelper::DidFinishNavigation(
   // documents.
   if (!navigation_handle->IsInMainFrame()) {
     DVLOG(3) << "FinishNavigation NOT in main frame";
+    is_same_document_navigation_ = false;
```

Comment:
This doesn't feel right - if the navigation is in a subframe, I don't think we want to change any state

Reply:
You're right, the main fix was in Cleanup and I had just included this as well because I mistakenly thought it should be reset in that case too.  Fixed and re-pushed.